### PR TITLE
fix: backwards compatibility for alerts

### PR DESCRIPTION
**Pull request by AdityaHegde**
Alerts created before 0.48 cannot be opened anymore. Also dashboard name is `undefined` even when created in the new release.

### CODE REVIEW

File: proto/buf.gen.ui.yaml [hunk 1]
```diff
@@ -4,6 +4,6 @@ managed:
   go_package_prefix:
     default: github.com/rilldata/rill
 plugins:
-  - plugin: buf.build/bufbuild/es
+  - plugin: buf.build/bufbuild/es:v1.10.0
```
Comment (esevastyanov) [inline proto/buf.gen.ui.yaml:7]: This change should not be passed as you've already passed generated proto sources.

File: proto/buf.gen.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ managed:
       - buf.build/googleapis/googleapis
       - buf.build/envoyproxy/protoc-gen-validate
 plugins:
-  - plugin: buf.build/protocolbuffers/go
+  - plugin: buf.build/protocolbuffers/go:v1.34.2
```
Comment (esevastyanov) [inline proto/buf.gen.yaml:10]: This change should not be passed too

**Review by esevastyanov — CHANGES_REQUESTED** [review]

File: proto/buf.gen.ui.yaml (hunk 1 above)
Comment (AdityaHegde) [reply proto/buf.gen.ui.yaml:7]: Anyone else generating will face the same issue if we dont pin the version.

**Review by nishantmonu51 — APPROVED** [review]
It's ok to pin versions for now. 
As a separate effort, we need to look into how would we upgrade proto versions in future without hitting the same issues again. I believe there should be some ways of adding library versions in the serialized proto objects, that the newer version can check to deserialize. 
@AdityaHegde : Can you help create an issue for tracking the upgrade effort ?

**Review by esevastyanov — APPROVED** [review]

File: proto/rill/runtime/v1/resources.proto [hunk 3]
```diff
@@ -365,6 +365,9 @@ message AlertSpec {
   int32 intervals_limit = 6;
   bool intervals_check_unclosed = 7;
   uint32 timeout_seconds = 8;
+  // for alerts that have not been edited since resolver and resolver_properties have been added
```
Comment (begelundmuller) [inline proto/rill/runtime/v1/resources.proto:368]: When the runtime restarts, it should parse all YAML files and update any resources that output a different proto object. So I don't think this should be necessary? Is the issue perhaps in some other place?

Comment (begelundmuller) [reply proto/rill/runtime/v1/resources.proto:368]: Here's the code that triggers a comparison of all resources when the parser first runs: https://github.com/rilldata/rill/blob/main/runtime/reconcilers/project_parser.go#L147

And here's the code that does the comparison for an alert object specifically: https://github.com/rilldata/rill/blob/main/runtime/reconcilers/project_parser.go#L520

Comment (AdityaHegde) [reply proto/rill/runtime/v1/resources.proto:368]: Good point. There were issues with stage runtimes not coming up. The root cause could have been that. (This PR was raised before it was identified)

**Pull request merged**